src/native/windows/apps/prunsrv/prunsrv.c

```diff
@@ -1834,7 +1837,17 @@ void WINAPI serviceMain(DWORD argc, LPTSTR *argv)
         /* Set console handler to capture CTRL events */
         SetConsoleCtrlHandler((PHANDLER_ROUTINE)console_handler, TRUE);
 
-        apxHandleWait(gWorker, INFINITE, FALSE);
+        if (SO_STOPTIMEOUT) {
+            /* we have a stop timeout */
+	    do {
+                /* wait 2 seconds */
+                apxHandleWait(gWorker, 2000, FALSE);
+            } while (!_exe_shutdown);
+            apxLogWrite(APXLOG_MARK_DEBUG "waiting %d sec... shutdown: %d", SO_STOPTIMEOUT, _exe_shutdown);
```

Comment:
This log statement seems wrong, we're not waiting at this point, we're _done_ waiting.

CC https://issues.apache.org/jira/browse/DAEMON-460